src/strands/models/anthropic.py

```diff
@@ -344,7 +344,7 @@ def format_chunk(self, event: dict[str, Any]) -> StreamEvent:
                 raise RuntimeError(f"event_type=<{event['type']} | unknown type")
 
     @override
-    def stream(self, request: dict[str, Any]) -> Iterable[dict[str, Any]]:
+    async def stream(self, request: dict[str, Any]) -> AsyncGenerator[dict[str, Any], None]:
```

Comment:
Is it worth making this a union and allowing a generator or an async generator?  That way we don't have to fake it for providers that aren't fully async - the SDK would handle that instead?

It would also simplify testing/mocking I think?

Reply:
As an added benefit, it would more easily let us identify what gaps we have with providers (those that aren't async would be more obvious)

Downside is that it's more work in the SDK, but IMHO I think it might be a valid tradeoff.  The SDK accepts both async/non-async and transforms it appropriately.  But it emits only async (and wrappers for non-async use cases)